roadmap.md

```diff
@@ -0,0 +1,57 @@
+## Overview
+The initial roadmap of CRI-O was lightweight and followed the main Kubernetes Container Runtime Interface (CRI) development lifecycle.
+This is partially because additional features on top of that are either integrated into a CRI-O release as soon as they’re ready, or are tracked through the Milestone mechanism in GitHub.
+Another reason is that feature availability is mostly tied to Kubernetes releases,
+and thus most of its long-term goals are already tracked in SIG-Node through the Kubernetes Enhancement Proposal (KEP) process.
+Finally, CRI-O’s long-term roadmap outside of features being added by SIG-Node is in part described by its mission:
+to be a secure, performant and stable implementation of the Container Runtime Interface.
+However, all of these together do construct a roadmap, and this document will describe how.
+
+## Milestones
+CRI-O’s primary internal planning mechanism is the Milestone feature in GitHub, along with Issues.
+Since CRI-O releases in lock-step with Kubernetes minor releases,
+where the CRI-O community aims to have a x.y.0 release released within three days after the corresponding Kubernetes x.y.0 release, there is a well established deadline that must be met.
+For PRs and Issues that are targeted at a particular x.y.0 release can be added to the x.y Milestone and they will be considered for priority in leading up to the actual release.
+However, since CRI-O’s releases are time bound and partially out of the CRI-O communities’ control, tagging a PR or issue with a Milestone does not guarantee it will be included.
+Users or contributors who don’t have permissions to add the Milestone can request an Approver do so.
+If there is disagreement, the standard [Conflict resolution and voting][github.com/cri-o/cri-o/tree/main/CONTRIBUTORS.md] mechanism will be used.
+Pull requests targeting patch release-x.y branches are not part of any milestone.
+The release branches are decoupled from the [Kubernetes patch release cycle](https://k8s.io/releases/patch-releases) and fixes can be merged ad-hoc.
+The support for patch release branches follows the yearly Kubernetes period and can be longer based on contributor bandwidth. 
+
+## SIG-Node
+CRI-O’s primary purpose is to be a CRI compliant runtime for Kubernetes, and thus most of the features that CRI-O adds are added to remain conformant to the CRI.
+Often, though not always, CRI-O will attempt to support new features in Kubernetes while they’re in the Alpha stage, though sometimes this target is missed and support is added while the feature is in Beta.
+To track the features that may be added to CRI-O from upstream, one can watch SIG-Node’s KEPs for a given release.
+If a particular feature interests you, the CRI-O community recommends you open an issue in CRI-O so it can be included in the Milestone for that given release.
+CRI-O maintainers are involved in SIG-Node in driving various upstream initiatives that can be tracked in the SIG-Node planning document.
```

Comment:
There are multiple references of SIG-Node, but no link to it. Maybe adding a link here (or at the first occurrence of "SIG-Node") would be nice to help track those initiatives?

Reply:
I added a couple of links embellishing references I made, PTAL @littlejawa :)